crates/tabby/src/services/chat.rs

```diff
@@ -99,32 +106,52 @@ impl ChatService {
         &self,
         request: &ChatCompletionRequest,
     ) -> BoxStream<ChatCompletionChunk> {
+        let mut log_output = vec![];
```

Comment:
consider create an helper struct `ChatCompletionEventBuilder` to handle the appending / event building logic

Reply:
Started doing that, it wasn't looking particularly helpful since the only real logic that needs to be pulled out is converting the message list, so I wrote a helper function for that instead.